### PR TITLE
Device search timings reduction and protocol downselection

### PR DESCRIPTION
**Pull request by themarpe**

### CODE REVIEW

**Review by szabi-luxonis — APPROVED** [review]
LGTM, apart from style changes so CI passes.

File: src/xlink/XLinkConnection.cpp [hunk 1]
```diff
@@ -126,12 +144,22 @@ DeviceInfo XLinkConnection::bootBootloader(const DeviceInfo& deviceInfo) {
...
     XLinkError_t rc;
+    auto bootupTimeout = WAIT_FOR_BOOTUP_TIMEOUT_USB;
+    if(deviceToWait.desc.protocol == X_LINK_TCP_IP){
+        bootupTimeout = WAIT_FOR_BOOTUP_TIMEOUT_TCPIP;
+    }
+
+    // Override with environment variables, if set
+    const std::vector<std::pair<std::string, std::chrono::milliseconds*>> evars = {
+        {"DEPTHAI_BOOTUP_TIMEOUT", &bootupTimeout},
+    };
```
Comment (alex-luxonis) [inline src/xlink/XLinkConnection.cpp:155]: Parsing of this one missing?

Comment (themarpe) [reply src/xlink/XLinkConnection.cpp:160]: Added

File: include/depthai/device/DeviceBase.hpp [hunk 2]
```diff
@@ -61,6 +61,13 @@ class DeviceBase {
 
     // static API
 
+    /**
+     * @brief Get the Default Search Time for finding devices
+     *
+     * @returns Default serach time in milliseconds
```
Comment (alex-luxonis) [inline include/depthai/device/DeviceBase.hpp:67]: > `serach`

typo

**Review by alex-luxonis — APPROVED** [review]

**Comment by szabi-luxonis** [comment]
When connected to debugger, first attempt to connect to the device fails after these changes with `X_LINK_DEVICE_NOT_FOUND`. Second one works.
Legacy debugging works for the first time.

**Comment by themarpe** [comment]
Thanks - will address

**Comment by themarpe** [comment]
@szabi-luxonis - fixed in 6283831f521d4022bc0e7fa8c019d0ec9aad1f52

**Pull request merged**